### PR TITLE
bottom tab bar & nav changes

### PR DESCRIPTION
**Pull request by sudoDeznit**
This is the first PR for the app with new design

**Changes made**
- Removed react navigation 4.x and added 5.x 
- configured the tab routes and other routes for new design
- rn vector icons was 2 years old and was missing lots of icons and even names were different from docs, So upgraded it to 8.x and removed other bundled icons from ios that were not added in the info.plist and kept only that are used (ionicons, materialicons)
- added placeholder screens for tab routes
- added a global fab that takes to the camera page
- removed RootContainerScreen and SwipeScreen -- both are not needed.

**Note**
- You will need to do pod install and do a fresh build.
- This is just the first PR so everything is not perfect.
- There are no big core changes. 
- This PR is big because this is the first one.. next ones will be smaller

### CODE REVIEW

**Comment by xlcrr** [comment]
We should merge the 2nd PR into this one. But `sudoDeznit:upgrade0.1` is not being picked up as a branch?

File: routes/tabRoutes.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                }
+            }}>
+            <Tab.Screen name="HOME" component={HomeScreen} />
+            <Tab.Screen name="STATS" component={StatsScreen} />
+            <Tab.Screen name="TEAM" component={TeamScreen} />
+            <Tab.Screen name="RANKING" component={RankingScreen} />
+            <Tab.Screen name="UPLOAD" component={LeftPage} />
+        </Tab.Navigator>
+        {/* fab will be shown on all screens which shows bottom tabbar */}
+        <Fab navigation={navigation} />
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply routes/tabRoutes.js:74]: FAB - Floating Action Button is the camera icon thats shown on all the screens

File: screens/auth/WelcomeScreen.js [hunk 2]
```diff
@@ -35,17 +35,10 @@ const SLIDE_DATA = [
...
-        props.checkForToken();
-    }
-
-    componentDidMount() {
-        this.props.checkForToken();
     }
 
     goToAuth(auth) {
-        this.props.navigation.navigate('auth', { auth });
+        this.props.navigation.navigate('AUTH');
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply screens/auth/WelcomeScreen.js:41]: I will make changes to this screen when I work on onboarding screens.
But what are the things to be fixed here ??

File: App.js [hunk 3]
```diff
@@ -29,16 +33,19 @@ const App = () => {
...
-            </PersistGate>
-        </Provider>
+        <NavigationContainer>
+            <Provider store={store}>
+                <PersistGate persistor={persistor}>
+                    <LanguageProvider
+                        language={lang}
+                        defaultLanguage={'en'}
+                        translations={langs}>
+                        {/* <RootContainer /> */}
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply App.js:43]: RootContainer page is removed in b32fcd56a55e05a286670e51b04e2db9344be2bc.

but the comment is still there which I will remove

[the remark this answers is not in the document]
Comment (xlcrr) [reply App.js:43]: If we don't need it anymore, delete it

File: routes/tabRoutes.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xlcrr) [reply routes/tabRoutes.js:74]: Can you add a comment that tells the next reader what this is?

File: screens/auth/WelcomeScreen.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (xlcrr) [reply screens/auth/WelcomeScreen.js:41]: I can't remember :D I will check again soon

**Comment by xlcrr** [comment]
Is this PR still a draft?

**Pull request merged**